xonsh/tools.py

```diff
@@ -1837,15 +1837,26 @@ def format_color(string, **kwargs):
     shell instances method of the same name. The results of this function should
     be directly usable by print_color().
     """
-    return builtins.__xonsh__.shell.shell.format_color(string, **kwargs)
+    try:
+        return builtins.__xonsh__.shell.shell.format_color(string, **kwargs)
+    except AttributeError:
```

Comment:
Can you use `if hasattr()' instead please? I believe it is faster than try-except

Reply:
I was curious about this one, so performed a quick test:
```python
class A():
    gotit = 1

a = A()

def t_try_happy():
    try:
        a.gotit
    except AttributeError:
        pass

def t_try_sad():
    try:
        a.notthere
    except AttributeError:
        pass

def t_hasattr_happy():
    if hasattr(a, "gotit"):
        a.gotit

def t_hasattr_sad():
    if hasattr(a, "notthere"):
        a.notthere
```

Results (cpython 3.8.5):
```
> timeit.timeit(t_try_happy, number=10000)
0.0008203879988286644
> timeit.timeit(t_try_sad, number=10000)
0.0037201289997028653
> timeit.timeit(t_hasattr_happy, number=10000)
0.0011529879993759096
> timeit.timeit(t_hasattr_sad, number=10000)
0.000985652000963455
```

So it seems that for the happy path, `try` is a little bit quicker. For the _unhappy_ it's much much slower indeed, but that should really be an edge case (using `print_color` without an initialized shell).

Of course, I can replace `try` with `hasattr` if you want me to, but I needed to check :)

Reply:
First off, thanks for the timing work! My concern here is that we will be meaning to hit the try-sad case (the slow case) in xonshrc (which is during startup). I am really trying to figure out how to keep startup times as low as possible, with the goal of getting to 100 ms.  Does this make sense?

Reply:
Sure, now it's more clear, will do the change.